core/modules/replica/HttpFileReaderApp.cc

```diff
@@ -0,0 +1,144 @@
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+// Class header
+#include "replica/HttpFileReaderApp.h"
+
+// System headers
+#include <fstream>
+#include <iostream>
+#include <stdexcept>
+#include <vector>
+
+// Qserv headers
+#include "replica/HttpFileReader.h"
+
+using namespace std;
+
+namespace {
+
+string const description =
+    "This application reads files from an object store over the HTTP/HTTPS protocol."
+    " If option '--file=<file>' is present the file's content will be writted into the"
+    " specified file. Otherwise the content will be printed onto the standard output stream.";
+
+} /// namespace
+
+namespace lsst {
+namespace qserv {
+namespace replica {
+
+HttpFileReaderApp::Ptr HttpFileReaderApp::create(int argc, char* argv[]) {
+    return Ptr(new HttpFileReaderApp(argc, argv));
+}
+
+
+HttpFileReaderApp::HttpFileReaderApp(int argc, char* argv[])
+    :   Application(
+            argc, argv,
+            ::description,
+            false   /* injectDatabaseOptions */,
+            false   /* boostProtobufVersionCheck */,
+            false   /* enableServiceProvider */
+        ) {
+
+    parser().required(
+        "url",
+        "The URL to read data from.",
+        _url
+    ).option(
+        "method",
+        "The HTTP method. Allowed values: GET, POST, PUT, DELETE.",
+        _method
+    ).option(
+        "header",
+        "The HTTP header to be sent with a request. Note this application allows"
+        " only one header.",
+        _header
+    ).option(
+        "data",
+        "The data to be sent in the body of a request.",
+        _data
+    ).flag(
+        "no-ssl-verify-peer",
+        "The flag that would disable verifying the peer's SSL certificate.",
+        _noSslVerifyPeer
+    ).option(
+        "ca-path",
+        "A path to a directory holding CA certificates to verify the peer with."
+        " This option is ignored if flag --no-ssl-verify-peer is specified.",
+        _caPath
+    ).option(
+        "ca-info",
+        "A path to a Certificate Authority (CA) bundle to verify the peer with."
+        " This option is ignored if flag --no-ssl-verify-peer is specified.",
+        _caInfo
+    ).option(
+        "file",
+        "A path to an output file where the content received from a remote source will be written."
+        "  If the option is not specified then the content will be printed onto the standard output"
+        " stream. This option is ignored if flag --silent is specified.",
+        _file
+    ).flag(
+        "silent",
+        "The flag that would disable printing or writing the content received from a remote"
+        " source.",
+        _silent
+    );
+}
+
+
+int HttpFileReaderApp::runImpl() {
+    string const context = "HttpFileReaderApp::" + string(__func__) + "  ";
+    vector<string> const allowedMethods = {"GET", "POST", "PUT", "DELETE"};
+    if (allowedMethods.cend() == find(allowedMethods.cbegin(), allowedMethods.cend(), _method)) {
```

Comment:
I think Python version of this would be shorter :)
In C++ you don't need `vector` to use algorithms, some thing simpler and more efficient like this would do:
```
    const char* allowedMethods[] = {"GET", "POST", "PUT", "DELETE"};
    if (std::count(std::cbegin(allowedMethods), std::cend(allowedMethods), _method) == 0) {
    }
```

Reply:
👍